### PR TITLE
basmens/issue45

### PR DESCRIPTION
**Pull request by basmens**

### CODE REVIEW

File: CellView.pde [hunk 1]
```diff
@@ -105,18 +105,20 @@ class CellView extends ViewBase {
                     vertex(50 - 0.15 * energySymbolSize, 50 - 0.15 * energySymbolSize);
                     endShape(CLOSE);
 
-                    if(screenSize > 75) {
+                    if(screenSize > 80) {
                         float progressToNextCodonTick = norm(cellModel.ticksSinceLastCodonTick, 0, cellModel.ticksPerCodonTick);
+                        ArrayList<CodonBaseModel> codonModels = cellModel.codonModels;
```
Comment (BenMens) [inline CellView.pde]: Do you mean to make a copy of codonModels. If so, you should write:

ArrayList<CodonBaseModel> codonModels = (ArrayList<CodonBaseModel>)cellModel.codonModels.clone();

File: CellView.pde [hunk 2]
```diff
@@ -141,6 +143,9 @@ class CellView extends ViewBase {
                         if (cellModel.codonModels.size() != 0) {
                             float previousCodonAngle = cellModel.codonModels.get(cellModel.previousExecuteHandPosition).segmentAngleInCodonCircle;
```
Comment (BenMens) [inline CellView.pde:144]: I still get a OutOfBounds exception. Processing doesn't show me the location, but I think it is here. I think you should apply a simular fix for this block.

**Review by BenMens — CHANGES_REQUESTED** [review]
Problem not solved yet. Please fix this.

**Comment by basmens** [comment]
can you please tell me how to replecate this bug. i ahve tried many runs, but not one of them encountered a problem.

**Review by BenMens — APPROVED** [review]
Thanks!

**Pull request merged**